Index: value.cpp
===================================================================
--- value.cpp	(revision 1907)
+++ value.cpp	(working copy)
@@ -209,7 +209,8 @@
     {
         std::vector<byte>::size_type end = value_.size();
         for (std::vector<byte>::size_type i = 0; i != end; ++i) {
-            os << static_cast<int>(value_[i]) << " ";
+            os << static_cast<int>(value_[i]);
+            if (i < end - 1) os << " ";
         }
         return os;
     }
