--- exiv2-0.25.orig/src/sonymn.cpp
+++ exiv2-0.25/src/sonymn.cpp
@@ -303,6 +357,9 @@
             else if (val == "3000") os << "ARW 2.0";
             else if (val == "3100") os << "ARW 2.1";
             else if (val == "3200") os << "ARW 2.2";
+            else if (val == "3300") os << "ARW 2.3";
+            else if (val == "3310") os << "ARW 2.3.1";
+            else if (val == "3320") os << "ARW 2.3.2";
             else                    os << "(" << value << ")";
         }
         return os;
