--- a/src/canonmn.cpp
+++ b/src/canonmn.cpp
@@ -864,6 +864,7 @@ namespace Exiv2 {
         { 150, "Sigma 30mm f/1.4 DC HSM"                                    }, // 2
         { 150, "Sigma 24mm f/1.8 DG Macro EX"                               }, // 3
         { 150, "Sigma 28mm f/1.8 DG Macro EX"                               }, // 4
+        { 150, "Sigma 18-35mm f/1.8 DC HSM | A"                             }, // 5
         { 151, "Canon EF 200mm f/2.8L"                                      },
         { 152, "Canon EF 300mm f/4L IS"                                     }, // 0
         { 152, "Sigma 12-24mm f/4.5-5.6 EX DG ASPHERICAL HSM"               }, // 1
