diff --git a/src/canonmn.cpp b/src/canonmn.cpp
index 1c0a73a..0326848 100644
--- a/src/canonmn.cpp
+++ b/src/canonmn.cpp
@@ -474,6 +474,7 @@ namespace Exiv2 {
         {  33, "Voigtlander or Zeiss Lens"                                  }, // 0
         {  33, "Voigtlander Ultron 40mm f/2 SLII Aspherical"                }, // 1
         {  33, "Zeiss Distagon 35mm T* f/2 ZE"                              }, // 2
+        {  33, "Carl Zeiss Planar T* 50mm f/1.4 ZE"                         }, // 3
         {  35, "Canon EF 35-80mm f/4-5.6"                                   },
         {  36, "Canon EF 38-76mm f/4.5-5.6"                                 },
         {  37, "Canon EF 35-80mm f/4-5.6"                                   }, // 0
