diff -urwEB exiv2-0.26.org/src/image.cpp exiv2-0.26/src/image.cpp
--- exiv2-0.26.org/src/image.cpp	2017-01-24 10:08:23.000000000 -0500
+++ exiv2-0.26/src/image.cpp	2017-02-13 14:04:44.719463129 -0500
@@ -392,6 +392,11 @@
                                 : 1
                                 ;
 
+                // Break for unknown tag types else we may get segfault.
+                if (type == 0) {
+                     break;
+                }
+
                 // if ( offset > io.size() ) offset = 0; // Denial of service?
                int buffer_size = size*count + pad;
                if (buffer_size < 4) buffer_size = 4;
