--- a/src/canonmn.cpp	2011-02-13 15:39:15.000000000 +0100
+++ b/src/canonmn.cpp	2011-05-25 19:58:40.420111647 +0200
@@ -549,6 +549,7 @@
         { 154, "Canon EF 20mm f/2.8 USM"                                    },
         { 155, "Canon EF 85mm f/1.8 USM"                                    },
         { 156, "Canon EF 28-105mm f/3.5-4.5 USM"                            },
+        { 156, "Tamron SP AF 70-300mm F4-5.6 Di VC USD"                     }, // 1
         { 160, "Canon EF 20-35mm f/3.5-4.5 USM"                             }, // 0
         { 160, "Tamron AF 19-35mm f/3.5-4.5"                                }, // 1
         { 161, "Canon EF 28-70mm f/2.8L"                                    }, // 0
@@ -671,6 +672,7 @@
         { 150, printCsLensByFocalLength },
         { 152, printCsLensByFocalLength },
         { 153, printCsLensByFocalLength },
+	{ 156, printCsLensByFocalLength },
         { 161, printCsLensByFocalLength },
         { 169, printCsLensByFocalLength },
         { 173, printCsLensByFocalLength }, // works partly
