Index: private.h
===================================================================
--- private.h	(revision 1357)
+++ private.h	(working copy)
@@ -130,15 +130,6 @@
 #endif /* !defined FILENAME_MAX */
 
 /*
-** SunOS 4.1.1 libraries lack remove.
-*/
-
-#ifndef remove
-extern int	unlink P((const char * filename));
-#define remove	unlink
-#endif /* !defined remove */
-
-/*
 ** Some ancient errno.h implementations don't declare errno.
 ** But some newer errno.h implementations define it as a macro.
 ** Fix the former without affecting the latter.
