Index: src/tags.cpp
===================================================================
--- src/tags.cpp	(revision 3631)
+++ src/tags.cpp	(working copy)
@@ -534,6 +534,9 @@
                 "unit is used for both <XResolution> and <YResolution>. If "
                 "the image resolution is unknown, 2 (inches) is designated."),
                 ifd0Id, imgStruct, unsignedShort, 1, printExifUnit),
+        TagInfo(0x0129, "PageNumber", N_("Page Number"),
+                N_("The page number of the page from which this image was scanned."),
+                ifd0Id, imgStruct, unsignedShort, 2, printValue),
         TagInfo(0x012d, "TransferFunction", N_("Transfer Function"),
                 N_("A transfer function for the image, described in tabular style. "
                 "Normally this tag is not necessary, since color space is "
Index: src/tiffimage.cpp
===================================================================
--- src/tiffimage.cpp	(revision 3631)
+++ src/tiffimage.cpp	(working copy)
@@ -2123,6 +2123,7 @@
             { 0x0124, ifd0Id }, // Exif.Image.T4Options
             { 0x0125, ifd0Id }, // Exif.Image.T6Options
             { 0x0128, ifd0Id }, // Exif.Image.ResolutionUnit
+            { 0x0129, ifd0Id }, // Exif.Image.PageNumber
             { 0x012d, ifd0Id }, // Exif.Image.TransferFunction
             { 0x013d, ifd0Id }, // Exif.Image.Predictor
             { 0x013e, ifd0Id }, // Exif.Image.WhitePoint
