Index: include/exiv2/exiv2.hpp
===================================================================
--- include/exiv2/exiv2.hpp	(revision 3510)
+++ include/exiv2/exiv2.hpp	(working copy)
@@ -45,7 +45,6 @@
 #include "gifimage.hpp"
 #include "image.hpp"
 #include "iptc.hpp"
-#include "http.hpp"
 #include "jp2image.hpp"
 #include "jpgimage.hpp"
 #include "metadatum.hpp"
Index: include/exiv2/http.hpp
===================================================================
--- include/exiv2/http.hpp	(revision 3510)
+++ include/exiv2/http.hpp	(working copy)
@@ -5,7 +5,7 @@
 #include <map>
 #include <algorithm>
 
-#include <exiv2/exiv2.hpp>
+#include "exiv2.hpp"
 
 namespace Exiv2 {
     typedef std::map<std::string,std::string> dict_t;
Index: src/http.cpp
===================================================================
--- src/http.cpp	(revision 3510)
+++ src/http.cpp	(working copy)
@@ -26,7 +26,7 @@
 // included header files
 #include "config.h"
 
-#include "exiv2.hpp"
+#include "http.hpp"
 
 #include <sys/types.h>
 #include <stdio.h>
