Index: convert.cpp
===================================================================
--- convert.cpp	(revision 1907)
+++ convert.cpp	(working copy)
@@ -590,6 +590,7 @@
 
             snprintf(buf, sizeof(buf), "%.9f", dsec);
             buf[sizeof(buf) - 1] = 0;
+            buf[1] = '.'; // some locales use ','
             subsec = buf + 1;
 
             Exiv2::ExifData::iterator datePos = exifData_->findKey(ExifKey("Exif.GPSInfo.GPSDateStamp"));
