Index: exiv2-0.22/src/canonmn.cpp
===================================================================
--- exiv2-0.22.orig/src/canonmn.cpp	2011-10-08 09:05:27.000000000 +0200
+++ exiv2-0.22/src/canonmn.cpp	2011-10-08 09:05:43.000000000 +0200
@@ -494,6 +494,7 @@
         {  49, "Canon EF-S 55-250mm f/4-5.6 IS"                             },
         {  50, "Canon EF-S 18-200mm f/3.5-5.6 IS"                           },
         {  51, "Canon EF-S 18-135mm f/3.5-5.6 IS"                           },
+        {  52, "Canon EF-S 18-55mm f/3.5-5.6 IS II"                         },
         {  94, "Canon TS-E 17mm f/4L"                                       },
         {  95, "Canon TS-E 24.0mm f/3.5 L II"                               },
         { 124, "Canon MP-E 65mm f/2.8 1-5x Macro Photo"                     },
