diff -ur exiv2-0.26-org/src/CMakeLists.txt exiv2-0.26/src/CMakeLists.txt
--- exiv2-0.26-org/src/CMakeLists.txt	2016-12-10 12:51:21.979012000 -0500
+++ exiv2-0.26/src/CMakeLists.txt	2016-12-10 09:55:44.000000000 -0500
@@ -254,6 +254,7 @@
 
 if ( MSVC )
     source_group("Header Files" FILES ${LIBEXIV2_HDR} )
+    TARGET_LINK_LIBRARIES( exiv2lib ${PRIVATE_VAR} ${ZLIB_LIBRARIES} )
     TARGET_LINK_LIBRARIES( exiv2lib ${PRIVATE_VAR} ${EXPAT_LIBRARIES} )
     source_group("Header Files" FILES ${LIBCURL_HDR} )
     TARGET_LINK_LIBRARIES( exiv2lib ${PRIVATE_VAR} ${CURL_LIBRARIES} )
