--- exiv2-0.26.svn.1/bootstrap.linux	2017-02-08 10:10:36.000000000 -0500
+++ exiv2-0.26.svn/bootstrap.linux	2017-02-27 16:40:41.999756070 -0500
@@ -19,7 +19,6 @@
       -DCMAKE_INSTALL_PREFIX=/usr \
       -DEXIV2_ENABLE_SHARED=ON \
       -DEXIV2_ENABLE_XMP=ON \
-      -DEXIV2_ENABLE_LIBXMP=ON \
       -DEXIV2_ENABLE_PNG=ON \
       -DEXIV2_ENABLE_NLS=ON \
       -DEXIV2_ENABLE_PRINTUCS2=ON \
--- exiv2-0.26.svn.1/bootstrap.macports	2017-02-08 10:10:35.000000000 -0500
+++ exiv2-0.26.svn/bootstrap.macports	2017-02-27 16:40:50.799902189 -0500
@@ -62,7 +62,6 @@
       -DCMAKE_MODULE_PATH="${INSTALL_PREFIX}/share/cmake/modules" \
       -DEXIV2_ENABLE_SHARED=ON \
       -DEXIV2_ENABLE_XMP=ON \
-      -DEXIV2_ENABLE_LIBXMP=ON \
       -DEXIV2_ENABLE_PNG=ON \
       -DEXIV2_ENABLE_NLS=ON \
       -DEXIV2_ENABLE_PRINTUCS2=ON \
