diff -Nurpd a/src/makernote.cpp b/src/makernote.cpp
--- a/src/makernote.cpp	2012-04-11 18:02:44.000000000 +0200
+++ b/src/makernote.cpp	2014-11-04 19:04:45.591750003 +0100
@@ -70,6 +70,7 @@ namespace Exiv2 {
         { "OLYMPUS",        ifdIdNotSet, newOlympusMn,   0               }, // mnGroup_ is not used
         { "Panasonic",      panasonicId, newPanasonicMn, newPanasonicMn2 },
         { "PENTAX",         ifdIdNotSet, newPentaxMn,    0               }, // mnGroup_ is not used
+        { "RICOH",          ifdIdNotSet, newPentaxMn,    0               }, // mnGroup_ is not used
         { "SAMSUNG",        samsung2Id,  newSamsungMn,   newSamsungMn2   },
         { "SIGMA",          sigmaId,     newSigmaMn,     newSigmaMn2     },
         { "SONY",           ifdIdNotSet, newSonyMn,      0               }, // mnGroup_ is not used
