Index: config/exiv2.pc.cmake
===================================================================
--- config/exiv2.pc.cmake	(revision 2329)
+++ config/exiv2.pc.cmake	(working copy)
@@ -6,6 +6,6 @@
 Name: exiv2
 Description: Exif and IPTC metadata library and tools
 Version: @PACKAGE_VERSION@
-URL: ${PACKAGE_URL}
+URL: @PACKAGE_URL@
 Libs: -L${libdir} -lexiv2 @EXV_LIB_STATIC@
 Cflags: -I${includedir}
Index: config/ConfigureChecks.cmake
===================================================================
--- config/ConfigureChecks.cmake	(revision 2329)
+++ config/ConfigureChecks.cmake	(working copy)
@@ -248,7 +248,7 @@
 ADD_CUSTOM_TARGET(uninstall "${CMAKE_COMMAND}" -P "${CMAKE_BINARY_DIR}/cmake_uninstall.cmake")
 
 IF( NOT MSVC )
-    CONFIGURE_FILE( config/exiv2.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/exiv2.pc )
+    CONFIGURE_FILE( config/exiv2.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/exiv2.pc @ONLY )
     INSTALL( FILES ${CMAKE_CURRENT_BINARY_DIR}/exiv2.pc DESTINATION lib/pkgconfig )
     CONFIGURE_FILE( config/exiv2.lsm.cmake ${CMAKE_CURRENT_BINARY_DIR}/exiv2.lsm)
     INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/exiv2.lsm DESTINATION lib/pkgconfig )
