--- a/src/nikonmn.cpp	Sun Jan 22 22:33:55 2012 +0000
+++ b/src/nikonmn.cpp	Wed Jan 25 22:36:24 2012 +0000
@@ -50,7 +50,7 @@
 #include <iomanip>
 #include <cassert>
 #include <cstring>
-#include <cmath>
+#include <math.h> //for log, pow, abs
 
 // *****************************************************************************
 // class member definitions
@@ -2454,8 +2454,8 @@
         std::ostringstream oss;
         oss.copyfmt(os);
         char sign = value.toLong() < 0 ? '-' : '+';
-        long h = long(abs(value.toLong())/60.0);
-        long min = abs(value.toLong()) - h*60;
+        long h = long(abs(value.toFloat())/60.0);
+        long min = abs(value.toFloat()) - h*60;
         os << std::fixed << "UTC " << sign << std::setw(2) << std::setfill('0') << h << ":" 
            << std::setw(2) << std::setfill('0') << min;
         os.copyfmt(oss);
