diff -Nurpd a/src/fujimn.cpp b/src/fujimn.cpp
--- a/src/fujimn.cpp	2016-04-17 13:54:31.067269928 +0100
+++ b/src/fujimn.cpp	2016-04-17 14:16:38.243454645 +0100
@@ -157,11 +157,17 @@ namespace Exiv2 {
 
     //! FilmMode, tag 0x1401
     extern const TagDetails fujiFilmMode[] = {
-        {    0, N_("F0/Standard")           },
-        {  256, N_("F1/Studio portrait")    },
-        {  512, N_("F2/Fujichrome")         },
-        {  768, N_("F3/Studio portrait Ex") },
-        { 1024, N_("F4/Velvia")             }
+	{    0, N_("F0/Standard (Provia)")				},
+	{  256, N_("F1/Studio Portrait")				},
+	{  272, N_("F1a/Studio Portrait Enhanced Saturation")		},
+	{  288, N_("F1b/Studio Portrait Smooth Skin Tone (Astia)")	},
+	{  304, N_("F1c/Studio Portrait Increased Sharpness")		},
+	{  512, N_("F2/Fujichrome (Velvia)")				},
+	{  768, N_("F3/Studio Portrait Ex")				},
+	{ 1024, N_("F4/Velvia")						},
+	{ 1280, N_("Pro Neg. Std")					},
+	{ 1281, N_("Pro Neg. Hi")					},
+	{ 1536, N_("Classic Chrome")					},
     };
 
     //! DynamicRange, tag 0x1402
