Index: src/properties.cpp
===================================================================
--- src/properties.cpp	(revision 3624)
+++ src/properties.cpp	(working copy)
@@ -402,8 +402,8 @@
     };
 
     extern const XmpPropertyInfo xmpLrInfo[] = {
-        { "hierarchicalsubject",    N_("HierarchicalSubject"),    "bag Text",  xmpBag,      xmpExternal, N_("Hierarchical Subject.")    },
-        { "privatertkinfo",         N_("PrivateRTKInfo"),         "Text",      xmpText,     xmpExternal, N_("Private RTK Info.")        },
+        { "hierarchicalSubject",    N_("HierarchicalSubject"),    "bag Text",  xmpBag,      xmpExternal, N_("Hierarchical Subject.")    },
+        { "privateRTKInfo",         N_("PrivateRTKInfo"),         "Text",      xmpText,     xmpExternal, N_("Private RTK Info.")        },
         // End of list marker
         { 0, 0, 0, invalidTypeId, xmpInternal, 0 }
     };
