diff -Nurpd lensfun-r2853/src/canonmn.cpp lensfun-r2853-canon/src/canonmn.cpp
--- lensfun-r2853/src/canonmn.cpp	2012-08-21 19:45:02.013828940 +0200
+++ lensfun-r2853-canon/src/canonmn.cpp	2012-08-21 19:50:59.017828642 +0200
@@ -669,7 +669,9 @@ namespace Exiv2 {
         { 491, "Canon EF 300mm f/2.8L IS II USM"                            },
         { 494, "Canon EF 600mm f/4.0L IS II USM"                            },
         { 495, "Canon EF 24-70mm f/2.8L II USM"                             },
-        { 4144,"Canon EF 40mm f/2.8 STM"                                    }
+        { 4143,"Canon EF-M 18-55mm f/3.5-5.6 IS STM"                        },
+        { 4144,"Canon EF 40mm f/2.8 STM"                                    },
+        { 4145,"Canon EF-M 22mm f/2 STM"                                    }
     };
 
     //! A lens id and a pretty-print function for special treatment of the id.
