From 3a7aa6ea5d35ec5fb147c5e8d947e50404b857ab Mon Sep 17 00:00:00 2001
From: Michael Pratt <michael@pratt.im>
Date: Sat, 5 Jul 2014 15:52:24 -0700
Subject: [PATCH] Run svn_version.sh with shell

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index 0295e91..d64912c 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -224,7 +224,7 @@ exv_conf.h: $(top_srcdir)/config/config.h
 
 # svn_version.h is only rewritten when the svn info | grep Revision has changed
 svn_version.h:
-	svn_version.sh
+	$(SHELL) svn_version.sh
 	
 version.hpp : svn_version.h
 	
-- 
2.0.0

