Index: include/exiv2/config.h
===================================================================
--- include/exiv2/config.h	(Revision 4761)
+++ include/exiv2/config.h	(Arbeitskopie)
@@ -6,7 +6,6 @@
 
 ///// Start of Visual Studio Support /////
 #ifdef  _MSC_VER
-#include <windows.h>
 
 #define _MSC_VER_2015 1900
 #define _MSC_VER_2013 1800
Index: include/exiv2/rwlock.hpp
===================================================================
--- include/exiv2/rwlock.hpp	(Revision 4761)
+++ include/exiv2/rwlock.hpp	(Arbeitskopie)
@@ -22,7 +22,9 @@
 #ifndef RW_LOCK_HPP
 #define RW_LOCK_HPP
 
-#ifndef _MSC_VER
+#ifdef _MSC_VER
+#include <windows.h>
+#else
 #include <pthread.h>
 #endif
 
