--- 0.21/src/preview.cpp	2010-11-06 22:57:15.000000000 +0800
+++ 0.21-bug745/src/preview.cpp	2010-12-06 10:39:49.000000000 +0800
@@ -248,7 +248,7 @@
         { 0,                   createLoaderExifDataJpeg, 5 },
         { 0,                   createLoaderExifDataJpeg, 6 },
         { 0,                   createLoaderExifDataJpeg, 7 },
-        { "image/x-raw",       createLoaderExifDataJpeg, 8 },
+        { "image/x-panasonic-rw2", createLoaderExifDataJpeg, 8 },
         { 0,                   createLoaderExifDataJpeg, 9 },
         { 0,                   createLoaderTiff,         0 },
         { 0,                   createLoaderTiff,         1 },
