--- canonmn.cpp.myorg	2014-03-17 13:33:09.000000000 +0200
+++ canonmn.cpp	2014-03-17 17:50:26.000000000 +0200
@@ -795,6 +795,7 @@
         { 213, "Canon EF 90-300mm f/4.5-5.6 USM"                            },
         { 214, "Canon EF-S 18-55mm f/3.5-5.6 USM"                           },
         { 215, "Canon EF 55-200mm f/4.5-5.6 II USM"                         },
+        { 217, "Tamron AF 18-270mm f/3.5-6.3 Di II VC PZD"                  },
         { 224, "Canon EF 70-200mm f/2.8L IS"                                },
         { 225, "Canon EF 70-200mm f/2.8L IS + 1.4x"                         },
         { 226, "Canon EF 70-200mm f/2.8L IS + 2x"                           },
