Index: src/CMakeLists.txt
===================================================================
--- src/CMakeLists.txt	(revision 3358)
+++ src/CMakeLists.txt	(working copy)
@@ -63,7 +63,6 @@
                           preview.hpp
                           properties.hpp
                           psdimage.hpp
-                          svn_version.h
                           quicktimevideo.hpp
                           rafimage.hpp
                           riffvideo.hpp
@@ -149,6 +148,12 @@
     ENDIF( ZLIB_FOUND )
 ENDIF( EXIV2_ENABLE_PNG )
 
+IF( NOT MSVC )
+    SET( LIBEXIV2_HDR     ${LIBEXIV2_HDR}
+                          svn_version.h
+    )
+ENDIF()
+
 ##
 # add xmp source if xmp is requested and xmp library not built
 IF( EXIV2_ENABLE_XMP )
