Index: include/exiv2/basicio.hpp
===================================================================
--- include/exiv2/basicio.hpp	(revision 4630)
+++ include/exiv2/basicio.hpp	(working copy)
@@ -559,6 +559,8 @@
         // Pimpl idiom
         class Impl;
         Impl* p_;
+        
+        long totalRead_;
 
     }; // class FileIo
 
Index: include/exiv2/config.h
===================================================================
--- include/exiv2/config.h	(revision 4630)
+++ include/exiv2/config.h	(working copy)
@@ -214,7 +214,10 @@
 #endif
 //////////////////////////////////////
 
+#undef EXV_HAVE_MMAP
+#undef EXV_HAVE_MUNMAP
 
+
 # include <stdio.h>
 #ifdef   EXV_HAVE_UNISTD_H
 #include <unistd.h>
Index: src/basicio.cpp
===================================================================
--- src/basicio.cpp	(revision 4630)
+++ src/basicio.cpp	(working copy)
@@ -351,6 +351,7 @@
 
     FileIo::FileIo(const std::string& path)
         : p_(new Impl(path))
+        , totalRead_(0)
     {
     }
 
@@ -1048,6 +1049,7 @@
             if (std::fclose(p_->fp_) != 0) rc |= 1;
             p_->fp_= 0;
         }
+        std::cout << "totalRead_ = " << totalRead_ << std::endl;
         return rc;
     }
 
@@ -1064,6 +1066,7 @@
     {
         assert(p_->fp_ != 0);
         if (p_->switchMode(Impl::opRead) != 0) return 0;
+        totalRead_ += rcount;
         return (long)std::fread(buf, 1, rcount, p_->fp_);
     }
 
