diff -rupN exiv2-0.18/src/pngchunk.cpp exiv2-0.18-new/src/pngchunk.cpp
--- exiv2-0.18/src/pngchunk.cpp	2008-12-15 16:24:37.000000000 +0000
+++ exiv2-0.18-new/src/pngchunk.cpp	2009-03-09 20:56:49.890451126 +0000
@@ -619,6 +619,9 @@ namespace Exiv2 {
                                             0,0,0,0,0,0,0,0,0,0, 0,0,0,0,0,0,0,0,0,0,
                                             0,0,0,0,0,0,0,0,0,0, 0,0,0,0,0,0,0,10,11,12,
                                             13,14,15};
+	if(text.pData_ == 0) {
+	  return DataBuf();
+	}
 
         sp = (char*)text.pData_+1;
 
