--- a/src/jpgimage.cpp	Sun Jan 22 22:33:55 2012 +0000
+++ b/src/jpgimage.cpp	Wed Jan 25 22:26:39 2012 +0000
@@ -793,7 +794,7 @@
             }
             else if (   skipApp1Exif == count
                      || skipApp1Xmp  == count
-                     || find(skipApp13Ps3.begin(), skipApp13Ps3.end(), count) != skipApp13Ps3.end()
+                     || std::find(skipApp13Ps3.begin(), skipApp13Ps3.end(), count) != skipApp13Ps3.end()
                      || skipCom      == count) {
                 --search;
                 io_->seek(size-bufRead, BasicIo::cur);
