Index: xmpsdk/CMakeLists.txt
===================================================================
--- xmpsdk/CMakeLists.txt	(revision 2890)
+++ xmpsdk/CMakeLists.txt	(working copy)
@@ -14,15 +14,15 @@
     IF(NOT MSVC AND NOT CYGWIN AND NOT MSYS)
         SET_SOURCE_FILES_PROPERTIES(${_currentfile} PROPERTIES COMPILE_FLAGS "-fPIC")
     ENDIF(NOT MSVC AND NOT CYGWIN AND NOT MSYS)
-ENDFOREACH(_currentfile ${XMPSRC})
+ENDFOREACH()
 
 IF( EXIV2_ENABLE_XMP AND EXIV2_ENABLE_LIBXMP )
-    ADD_LIBRARY( xmp STATIC ${XMPSRC} ${EXPAT_LIBRARIES} )
+    ADD_LIBRARY( xmp STATIC ${XMPSRC} )
     GET_TARGET_PROPERTY( XMPLIB xmp LOCATION )
     TARGET_LINK_LIBRARIES(xmp ${EXPAT_LIBRARIES})
     # SET( XMPLIB ${XMPLIB} PARENT_SCOPE )
     # INSTALL(TARGETS xmp ${INSTALL_TARGET_STANDARD_ARGS} )
-ENDIF( EXIV2_ENABLE_XMP AND EXIV2_ENABLE_LIBXMP )
+ENDIF()
 
 # That's all Folks!
 ##
