Index: src/tiffimage.cpp
===================================================================
--- src/tiffimage.cpp	(revision 3732)
+++ src/tiffimage.cpp	(working copy)
@@ -345,12 +345,11 @@
     // http://stackoverflow.com/questions/2342162/stdstring-formatting-like-sprintf
     static std::string stringFormat(const std::string fmt_str, ...) {
 		int n = ((int)fmt_str.size()) * 2; /* Reserve two times as much as the length of the fmt_str */
-		std::unique_ptr<char[]> formatted;
+		char* formatted = new char[n];
 		std::string str;
 		va_list     ap;
 		bool        ok = true;
 		while(ok) {
-			formatted.reset(new char[n]); /* Wrap the plain char array into the unique_ptr */
 			strcpy(&formatted[0], fmt_str.c_str());
 			va_start(ap, fmt_str);
 			int final = vsnprintf(&formatted[0], n, fmt_str.c_str(), ap);
@@ -358,7 +357,9 @@
 			ok = final < 0 || final >= n;
 			if ( ok ) n += abs(final - n + 1);
 		}
-		return std::string(formatted.get());
+		str = std::string(formatted);
+		delete[] formatted;
+		return str;
     }
 
     std::string stringValue(byte* buff,size_t size)
