From c19cb0c0d7b5821d30283348eb16a7d726b01fca Mon Sep 17 00:00:00 2001
From: draekko <draekko.software@gmail.com>
Date: Sat, 24 Jun 2017 10:41:26 -0400
Subject: [PATCH 2/2] Fix for bootstrap scripts with new codebase

---
 bootstrap.linux    | 1 -
 bootstrap.macports | 1 -
 2 files changed, 2 deletions(-)

diff --git a/bootstrap.linux b/bootstrap.linux
index 3502f6fc..55f54f6f 100755
--- a/bootstrap.linux
+++ b/bootstrap.linux
@@ -19,7 +19,6 @@ cmake -G "$MAKEFILES_TYPE" . \
       -DCMAKE_INSTALL_PREFIX=/usr \
       -DEXIV2_ENABLE_SHARED=ON \
       -DEXIV2_ENABLE_XMP=ON \
-      -DEXIV2_ENABLE_LIBXMP=ON \
       -DEXIV2_ENABLE_PNG=ON \
       -DEXIV2_ENABLE_NLS=ON \
       -DEXIV2_ENABLE_PRINTUCS2=ON \
diff --git a/bootstrap.macports b/bootstrap.macports
index 9ce03570..3decaa47 100755
--- a/bootstrap.macports
+++ b/bootstrap.macports
@@ -62,7 +62,6 @@ cmake -G "$MAKEFILES_TYPE" . \
       -DCMAKE_MODULE_PATH="${INSTALL_PREFIX}/share/cmake/modules" \
       -DEXIV2_ENABLE_SHARED=ON \
       -DEXIV2_ENABLE_XMP=ON \
-      -DEXIV2_ENABLE_LIBXMP=ON \
       -DEXIV2_ENABLE_PNG=ON \
       -DEXIV2_ENABLE_NLS=ON \
       -DEXIV2_ENABLE_PRINTUCS2=ON \
-- 
2.11.0

