Index: test/Makefile
===================================================================
--- test/Makefile	(revision 3812)
+++ test/Makefile	(working copy)
@@ -77,7 +77,8 @@
          write-test.sh        \
          write2-test.sh       \
          xmpparser-test.sh    \
-         conversions.sh
+         conversions.sh       \
+         iso65k-test.sh
 
 # video tests
 TESTV  = video-test.sh         
