Index: samples/addmoddel.cpp
===================================================================
--- samples/addmoddel.cpp	(revision 3351)
+++ samples/addmoddel.cpp	(working copy)
@@ -63,7 +63,7 @@
     std::string date = tag.toString();
     date.replace(0, 4, "2000");
     tag.setValue(date);
-    std::cout << "Modified key \"" << key
+    std::cout << "Modified key \"" << tag.key()
               << "\", new value \"" << tag.value() << "\"\n";
 
     // Alternatively, we can use findKey()
