Index: src/image.cpp
===================================================================
--- src/image.cpp	(revision 4647)
+++ src/image.cpp	(working copy)
@@ -614,6 +614,7 @@
     void Image::setIccProfile(Exiv2::DataBuf& iccProfile,bool bTestValid)
     {
         if ( bTestValid ) {
+            if ( iccProfile.pData_ && ( iccProfile.size_ < sizeof(long)) ) throw Error(53);
             long size = iccProfile.pData_ ? getULong(iccProfile.pData_, bigEndian): -1;
             if ( size!= iccProfile.size_ ) throw Error(53);
         }
