--- config/config.h.in.orig	Tue Jan 31 20:59:39 2012
+++ config/config.h.in	Tue Jan 31 21:00:41 2012
@@ -217,7 +217,7 @@
 #   define EXV_EXPORT __attribute__ ((visibility("default")))
 #   define EXV_DLLLOCAL __attribute__ ((visibility("hidden")))
 #   define EXV_DLLPUBLIC __attribute__ ((visibility("default")))
-#  elif defined(__SUNPRO_C) && (__SUNPRO_C >= 0x550)
+#  elif defined(__SUNPRO_C) || defined(__SUNPRO_CC)
 #   define EXV_IMPORT __global
 #   define EXV_EXPORT __global
 #   define EXV_DLLLOCAL __hidden
