Index: src/tags.cpp
===================================================================
--- src/tags.cpp	(revision 4439)
+++ src/tags.cpp	(working copy)
@@ -2405,8 +2405,8 @@
                 if (value.toRational(n).first != 0) break;
             }
             for (int i = 0; i < n + 1; ++i) {
-                const int32_t z = value.toRational(i).first;
-                const int32_t d = value.toRational(i).second;
+                const uint32_t z = value.toRational(i).first;
+                const uint32_t d = value.toRational(i).second;
                 if (d == 0)
                 {
                     os << "(" << value << ")";
