Index: src/types.cpp
===================================================================
--- src/types.cpp	(Revision 2327)
+++ src/types.cpp	(Arbeitskopie)
@@ -152,8 +152,10 @@
     {
         if (size > size_) {
             delete[] pData_;
+            pData_ = 0;
+            size_ = 0;
+            pData_ = new byte[size];
             size_ = size;
-            pData_ = new byte[size];
         }
     }
 
