Index: nikonmn.cpp
===================================================================
--- nikonmn.cpp	(revision 2666)
+++ nikonmn.cpp	(working copy)
@@ -2478,7 +2478,7 @@
         oss.copyfmt(os);
         char sign = value.toLong() < 0 ? '-' : '+';
         long h = long(abs(value.toFloat())/60.0);
-        long min = abs(value.toFloat()) - h*60;
+        long min = long(abs(value.toFloat()) - h*60);
         os << std::fixed << "UTC " << sign << std::setw(2) << std::setfill('0') << h << ":" 
            << std::setw(2) << std::setfill('0') << min;
         os.copyfmt(oss);
