--- psdimage.hpp_r1750	2009-02-28 20:41:11.000000000 +0100
+++ psdimage.hpp	2009-02-23 21:19:08.000000000 +0100
@@ -137,6 +137,9 @@
           @return 4 if opening or writing to the associated BasicIo fails
          */
         EXV_DLLLOCAL void doWriteMetadata(BasicIo& oIo);
+        EXV_DLLLOCAL uint32_t _setIptcData(const IptcData& iptcData, BasicIo& out);
+        EXV_DLLLOCAL uint32_t _setExifData(const ExifData& exifData, BasicIo& out);
+        EXV_DLLLOCAL uint32_t _setXmpData(const XmpData& xmpData, BasicIo& out);
         //@}
 
     }; // class PsdImage
