Index: src/convert.cpp
===================================================================
--- src/convert.cpp	(revision 1495)
+++ src/convert.cpp	(working copy)
@@ -1079,7 +1079,8 @@
                 continue;
             }
             // Todo: Escape non-ASCII characters in XMP text values
-            Iptcdatum id(IptcKey(to), 0);
+            IptcKey key(to);
+            Iptcdatum id(key);
             id.setValue(value);
             iptcData_->add(id);
         }
