Index: src/tags.cpp
===================================================================
--- src/tags.cpp	(revision 3889)
+++ src/tags.cpp	(working copy)
@@ -2464,16 +2464,6 @@
     std::ostream& print0x829a(std::ostream& os, const Value& value, const ExifData*)
     {
         Rational t = value.toRational();
-        if (t.first > 1 && t.second > 1 && t.second >= t.first) {
-            t.second = static_cast<uint32_t>(
-                static_cast<float>(t.second) / t.first + 0.5);
-            t.first = 1;
-        }
-        if (t.second > 1 && t.second < t.first) {
-            t.first = static_cast<uint32_t>(
-                static_cast<float>(t.first) / t.second + 0.5);
-            t.second = 1;
-        }
         if (t.second == 1) {
             os << t.first << " s";
         }
