diff -Nurpd a/src/canonmn.cpp b/src/canonmn.cpp
--- a/src/canonmn.cpp	2013-01-29 22:03:48.712014152 +0100
+++ b/src/canonmn.cpp	2013-01-29 22:05:07.820406431 +0100
@@ -671,7 +671,9 @@ namespace Exiv2 {
         { 494, "Canon EF 600mm f/4.0L IS II USM"                            },
         { 495, "Canon EF 24-70mm f/2.8L II USM"                             },
         { 505, "Canon EF 35mm f/2 IS USM"                                   },
-		{ 4144,"Canon EF 40mm f/2.8 STM"                                    }
+        { 4143,"Canon EF-M 18-55mm f/3.5-5.6 IS STM"                        },
+        { 4144,"Canon EF 40mm f/2.8 STM"                                    },
+        { 4145,"Canon EF-M 22mm f/2 STM"                                    }
     };
 
     //! A lens id and a pretty-print function for special treatment of the id.
