diff -Nurpd exiv2-r3052-a/src/samsungmn.cpp exiv2-r3052-b/src/samsungmn.cpp
--- exiv2-r3052-a/src/samsungmn.cpp	2013-06-15 14:56:07.447832215 +0200
+++ exiv2-r3052-b/src/samsungmn.cpp	2013-06-15 15:08:49.111831580 +0200
@@ -60,6 +60,7 @@ namespace Exiv2 {
         {  7, N_("Samsung NX 60mm F2.8 Macro ED OIS SSA") },
         {  8, N_("Samsung NX 16mm F2.4 Pancake")          },
         {  9, N_("Samsung NX 85mm F1.4 ED SSA")           },
+        { 10, N_("Samsung NX 45mm F1.8")                  },
         { 12, N_("Samsung NX 12-24mm F4-5.6 ED")          }
     };
 
