--- a/src/canonmn.cpp
+++ b/src/canonmn.cpp
@@ -1058,6 +1058,7 @@ namespace Exiv2 {
         { 4156,"Canon EF 50mm f/1.8 STM"                                    },
         { 4157,"Canon EF-M 18-150mm 1:3.5-6.3 IS STM"                       },
         { 4158,"Canon EF-S 18-55mm f/4-5.6 IS STM"                          },
+        { 4160,"Canon EF-S 35mm f/2.8 Macro IS STM"                         },
         {36910,"Canon EF 70-300mm f/4-5.6 IS II USM"                        },
         {36912,"Canon EF-S 18-135mm f/3.5-5.6 IS USM"                       },
         {65535,"n/a"                                                        }
