--- src/pentaxmn.cpp.orig	2013-05-05 09:43:57.000000000 +0200
+++ src/pentaxmn.cpp	2013-05-05 09:45:21.000000000 +0200
@@ -685,6 +685,7 @@
         { 0x07f2, "smc PENTAX-DA* 16-50mm F2.8 ED AL [IF] SDM (SDM not used)" },
         { 0x07f3, "smc PENTAX-DA 70mm F2.4 Limited" },
         { 0x07f4, "smc PENTAX-DA 21mm F3.2 AL Limited" },
+        { 0x08d7, "smc PENTAX-DA 18-135mm F3.5-5.6 ED AL [IF] DC WR" },
         { 0x08e2, "smc PENTAX-DA* 55mm F1.4 SDM" },
         { 0x08e3, "smc PENTAX DA* 60-250mm F4 [IF] SDM"},
         { 0x08e8, "smc PENTAX-DA 17-70mm F4 AL [IF] SDM" },
